pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -536,7 +544,7 @@ stdenv.mkDerivation ({
         exit 1
       fi
     ''}
-    export GHC_PACKAGE_PATH="$packageConfDir:"
+    ${lib.optionalString (!useCabalInstall) ''export GHC_PACKAGE_PATH="$packageConfDir:"''}
```

Comment:
Wouldn't this env var simply be ignored if using cabal-install? Why put it behind a switch?

Reply:
The `cabal` V2 CLI complains about it if you set it.  It will tell you to instead set `--package-db` to the same value (which the Nix build already does)